app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -250,6 +250,8 @@ def export
 
       fail HttpErrors::BadRequest, _("Repository content type must be 'yum' to export.") unless @repository.content_type == 'yum'
 
+      fail HttpErrors::BadRequest, _("On-demand repositories cannot be exported.") if @repository.download_policy == ::Runcible::Models::YumImporter::DOWNLOAD_ON_DEMAND
```

Comment:
nitpick: Since hammer/UI refer to this as `on demand`, can we remove the hyphen?